src/matter/mdns/MdnsBroadcaster.ts

```diff
@@ -45,7 +45,11 @@ export class MdnsBroadcaster implements Broadcaster {
             const ipMac = this.network.getIpMac(netInterface);
             if (ipMac === undefined) return [];
             const { mac, ips } = ipMac;
+            if( ips.length === 0) {
```

Comment:
I think we should sort out the inzterface before if we sorted out all IPs, so this would be not needed here. see below

Reply:
Upon conducting tests on a more complex network and desktop with multiple valid network adapters, I have discovered that the MdnsBroadcaster library functions in such a manner that it advertises multiple Matter devices instead of a singular device. In light of this discovery, I am currently delving into the specifications in order to ascertain the extent of this issue and if it must be addressed.

Reply:
☝️above is currently not an issue

Reply:
Cooland good to now :-)

Did you checked if "sorting the resulting IPS" makes any difference?